app/controllers/class_memberships_controller.rb

```diff
@@ -0,0 +1,42 @@
+class ClassMembershipsController < ApplicationController
+
+  # def new
+  #   @class_membership = ClassMembership.new
+  #   authorize @class_membership
+
+  # end
+
+  def index
+    @class_memberships = policy_scope(ClassMembership.includes(:user, :taught_class)).where(user: current_user)
+    authorize @class_memberships
+  end
+
+
+  def create
+    @taught_class = TaughtClass.find(params[:taught_class_id])
```

Comment:
Not sure about the names here. When you're talking to a friend are you also say class membership or do you use a different terminology. Try to apply ubiquitous language from the DDD book here.

> Ubiquitous Language is the term Eric Evans uses in Domain Driven Design for the practice of building up a common, rigorous language between developers and users.